Client refactor: code reformatting

sechkova opened the pull request:
**Description of the changes being introduced by the pull request**:

- Move file download related modules inside client directory
- Remove (un)safe_download functions
- Apply black and isort to the newly moved files
- Apply the new pylint config from api/pylintrc and fix the various linter issues
- Remove the use of six and future and make the rest of the imports vendoring-compatible

**Please verify and check that the pull request fulfills the following
requirements**:

- [ ] The code follows the [Code Style Guidelines](https://github.com/secure-systems-lab/code-style-guidelines#code-style-guidelines)
- [ ] Tests have been added for the bug fix or new feature
- [ ] Docs have been added for the bug fix or new feature

sechkova committed:
Move download modules inside client directory

The modules performing network download are used only by
the client side of TUF. Move them inside the client
directory for the refactored client.

Move the _mirror_*download functions from Updater to
mirrors.py.

Signed-off-by: Teodora Sechkova <tsechkova@vmware.com>

sechkova committed:
Remove (un)safe_download functions

The two functions safe/unsafe_download differ only by setting
a single boolean flag. Remove them and call directly
_download_file instead.

Signed-off-by: Teodora Sechkova <tsechkova@vmware.com>

sechkova committed:
Reformat client code

Run black and isort over the old modules which were
moved inside the client directory:
- download.py
- fetcher.py
- mirrors.py
- requests_fetcher.py

Signed-off-by: Teodora Sechkova <tsechkova@vmware.com>

sechkova committed:
Fix pylint C0301: Line too long

Fix line lenght exceeding 80 characters since the
black formatter does not wrap lines inside comments.

Signed-off-by: Teodora Sechkova <tsechkova@vmware.com>

sechkova committed:
Fix pylint W0212: protected-access

Fix "Access to a protected member _mirror_target_download,
_mirror_meta_download of a client class"

Signed-off-by: Teodora Sechkova <tsechkova@vmware.com>

sechkova committed:
Fix pylint C0103 and W0612

- fix C0103 invalid argument name STRICT_REQUIRED_LENGTH
- use 'dummy' as an accepted by pylint unused variable
name (W0612)

Signed-off-by: Teodora Sechkova <tsechkova@vmware.com>

sechkova committed:
Fix pylint R1720: Unnecessary "else" after "raise"

Signed-off-by: Teodora Sechkova <tsechkova@vmware.com>

sechkova committed:
Fix pylint W0703: broad-except

pylint cannot figure out that we store the exceptions
in a dictionary to raise them later so we disable
the warning. This should be reviewed in the future still.

Signed-off-by: Teodora Sechkova <tsechkova@vmware.com>

sechkova committed:
Remove use of future and six

six and future are Python 2 compatibility modules which are
no longer needed after the end of Python 2 support.

Signed-off-by: Teodora Sechkova <tsechkova@vmware.com>

sechkova committed:
Fix client imports

Fix imports to be vendoring compatible.

Signed-off-by: Teodora Sechkova <tsechkova@vmware.com>

jku commented on tuf/client_rework/updater_rework.py:
```diff
@@ -147,71 +148,22 @@ def download_target(self, target: Dict, destination_directory: str):
+            target, self._mirrors, self._fetcher
+        ):
+
+            try:
```
commit message says it only moves code but this obviously changes things... We already have an issue on figuring out the download methods design (and fixing the bugs it has with multiple mirrors) so I'm not expecting this commit to fix everything -- but I also don't understand  what this change does?

sechkova replied:
Thanks for catching this, it was a mistake after cherry-picking this commit from another branch.
I had to rebase and force push to fix this. This is the new commit: 35587c4

jku commented on tuf/client_rework/download.py:
```diff
@@ -99,7 +99,8 @@ def download_file(url, required_length, fetcher, STRICT_REQUIRED_LENGTH=True):
     # slash.
     url = six.moves.urllib.parse.unquote(url).replace("\\", "/")
-    logger.info("Downloading: " + repr(url))
+    msg = f"Downloading: {url}"
+    logger.info(msg)
```
I disagree with the whole logging-not-lazy commit: Working around warning like this is not useful.
* if we want to use f-strings with logging, let's silence the warning in the linter config
* alternatively, let's just use %-style strings (I lean this way)

jku commented on tuf/client_rework/mirrors.py:
```diff
@@ -110,7 +99,7 @@ def get_list_of_mirrors(file_type, file_path, mirrors_dict):
 
     list_of_mirrors = []
-    for dummy, mirror_info in six.iteritems(mirrors_dict):
+    for dummy, mirror_info in mirrors_dict.items():
```
nit: dummy is not even required `for mirror_info in mirrors_dict.values()`

jku commented on tuf/client_rework/download.py:
```diff
@@ -23,24 +23,13 @@
 import logging
 import tempfile
 import timeit
+import urllib
```
`from urllib import parse` maybe -- and then use parse.unquote() etc. Current code only works by accident since some other module happens to import the right thing

jku reviewed (changes requested):
Thanks, looks mostly good to me:
* disagree with logging-not-lazy (we can discuss though)
* The *_target_download() functions still make me itch -- whatever we do here let's next concentrate on getting those fixed

sechkova committed:
Disable pylint W1201: logging-not-lazy

Disable pylint's "Use lazy % formatting in logging functions"
warning until a common logging approach is decided. See #1334.

Signed-off-by: Teodora Sechkova <tsechkova@vmware.com>

sechkova commented:
Force pushed the new commits. The differences are in 
- 35587c4, thanks for catching the error
- 2f0bbd0
- Dropped the "logging-not-lazy" commit and simply ignored the warning with 101ab3d 

I chose the lazy way of just ignoring the logging warning now, since there is already an issue capturing the need of a decision: #1334

jku reviewed (approved):
LGTM, thanks. 

The issues left IMO are:
* download implementation is fishy
* the internal file locations may not be final (I think they should be in _internal/ or something)
but we have these in writing already.

So let's go with this and see where we end up :)
